examples/mplot3d/wire3d_animation_demo.py

```diff
@@ -1,38 +1,45 @@
-from __future__ import print_function
 """
-A very simple 'animation' of a 3D plot
+A very simple 'animation' of a 3D plot.  See also rotate_axes3d_demo.
 """
+
+from __future__ import print_function
+
 from mpl_toolkits.mplot3d import axes3d
 import matplotlib.pyplot as plt
 import numpy as np
 import time
 
 
 def generate(X, Y, phi):
+    '''
+    Generates Z data for the points in the X, Y meshgrid and parameter phi.
+    '''
     R = 1 - np.sqrt(X**2 + Y**2)
     return np.cos(2 * np.pi * X + phi) * R
 
+
 fig = plt.figure()
 ax = fig.add_subplot(111, projection='3d')
 
+# Make the X, Y meshgrid.
 xs = np.linspace(-1, 1, 50)
 ys = np.linspace(-1, 1, 50)
 X, Y = np.meshgrid(xs, ys)
-Z = generate(X, Y, 0.0)
 
+# Set the z axis limits so they aren't recalculated each frame.
+ax.set_zlim(-1, 1)
```

Comment:
Using this example as a test script, I see the same behaviour with versions 1.5.2, 1.5.1, 1.5.0, 1.4.0, 1.2.0, and 1.0.0.  I haven't found a commit where this behaviour doesn't happen.